src/components/Dropdown/DropdownItem.tsx

```diff
@@ -1,37 +1,57 @@
-/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
-/* eslint-disable jsx-a11y/click-events-have-key-events */
-
+import { useListItem } from '@floating-ui/react';
 import type { ComponentProps, FC, PropsWithChildren } from 'react';
+import { useContext } from 'react';
 import { twMerge } from 'tailwind-merge';
 import type { DeepPartial } from '../../';
 import { useTheme } from '../../';
 import { mergeDeep } from '../../helpers/merge-deep';
+import type { ButtonBaseProps } from '../Button/ButtonBase';
+import { ButtonBase } from '../Button/ButtonBase';
+import { DropdownContext } from './Dropdown';
 
 export interface FlowbiteDropdownItemTheme {
   base: string;
   icon: string;
 }
 
-export interface DropdownItemProps extends PropsWithChildren, ComponentProps<'li'> {
+export interface DropdownItemProps extends PropsWithChildren, ButtonBaseProps {
   icon?: FC<ComponentProps<'svg'>>;
   onClick?: () => void;
   theme?: DeepPartial<FlowbiteDropdownItemTheme>;
 }
+interface Props extends DropdownItemProps, Record<string, unknown> {}
 
-export const DropdownItem: FC<DropdownItemProps> = ({
+export const DropdownItem: FC<Props> = ({
   children,
   className,
   icon: Icon,
   onClick,
   theme: customTheme = {},
   ...props
 }) => {
+  const { ref, index } = useListItem({ label: typeof children === 'string' ? children : undefined });
+  const { activeIndex, dismissOnClick, getItemProps, handleSelect } = useContext(DropdownContext);
+  const isActive = activeIndex === index;
+
   const theme = mergeDeep(useTheme().theme.dropdown.floating.item, customTheme);
 
   return (
-    <li className={twMerge(theme.base, className)} onClick={onClick} {...props}>
-      {Icon && <Icon className={theme.icon} />}
-      {children}
+    <li role="menuitem">
+      <ButtonBase
+        ref={ref}
+        className={twMerge(theme.base, className)}
```

Comment:
I think that this is not the `theme.base`, maybe this is something else...

Reply:
@rluders it actually is. I left the `li menuitem` more of as a container to hold the accessible role, then the user will be able to style the `ButtonBase` as they like or to change the base tag - and still have the container item as role menuitem

Reply:
Yeah, makes sense heheh I will add it then